newIDE/app/src/EventsFunctionsList/index.js

```diff
@@ -271,6 +271,27 @@ export default class EventsFunctionsList extends React.Component<Props, State> {
     this._editName(newEventsFunction);
   };
 
+  _duplicateEventsFunction = (
+    eventsFunction: gdEventsFunction,
+    newFunctionIndex: number
+  ) => {
+    const { eventsFunctionsContainer } = this.props;
+    const newName = newNameGenerator(eventsFunction.getName(), name =>
+      eventsFunctionsContainer.hasEventsFunctionNamed(name)
+    );
+    let newEventsFunction = eventsFunction.clone();
+    newEventsFunction.setName(newName);
+    newEventsFunction = eventsFunctionsContainer.insertEventsFunction(
+      newEventsFunction,
+      newFunctionIndex
+    );
```

Comment:
It's a bit strange that `insertEventsFunction`does a copy because here a copy is needed to change the name and another one to insert it.

Reply:
it is, but if it works like that for the moment, no real reason to change I guess...

Reply:
Careful there is a memory leak here

Reply:
(I'll let you find/understand it, because it's important to be able to recognise these ;) - you are right to find this code "weird" as it is. Look also at how InsertEventsFunction is working because you might not need the clone at all).

Reply:
>  it allows to have 2 functions with the same name in the same list

Exact, this is a shortcoming on the design of the API of the list holding the events function.
If the API was perfect, it would probably force you to pass a new name.